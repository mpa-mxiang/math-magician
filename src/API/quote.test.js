import React from 'react';
import { render, screen, waitForElementToBeRemoved } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Quote from './quote';

test('renders quote and author', async () => {
  render(<Quote />);

  await waitForElementToBeRemoved(() => screen.getByText(/loading/i), { timeout: 20000 });

  const errorElement = screen.queryByText(/error/i, { className: 'quote' });

  if (errorElement) {
    expect(errorElement).not.toBeInTheDocument();
  } else {
    const quoteTextElement = screen.queryByText(/Quote:*/, { className: 'quote' });
    const authorTextElement = screen.queryByText(/ - /);

    if (quoteTextElement) {
      expect(quoteTextElement).toBeInTheDocument();
      const quoteText = quoteTextElement.textContent;
      expect(quoteText).toBeTruthy();
      expect(quoteText).toMatchSnapshot();
    }

    if (authorTextElement) {
      expect(authorTextElement).toBeInTheDocument();
      const authorText = authorTextElement.textContent;
      expect(authorText).toBeTruthy();
      expect(authorText).toMatchSnapshot();
    }
  }
});
