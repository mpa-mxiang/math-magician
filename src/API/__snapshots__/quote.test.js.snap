// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders quote and author 1`] = `"Quote: I started on an Apple II, which I had bought at the very end of 1978 for half of my annual income. I made $4,500 a year, and I spent half of it on the computer. -Bill Budge"`;

exports[`renders quote and author 2`] = `"Quote: I think I was lucky to come of age in a place and time - the American South in the 1960s and '70s - when the machine hadn't completely taken over life. The natural world was still the world, and machines - TV, telephone, cars - were still more or less ancillary, and computers were unheard of in everyday life. -Ben Fountain"`;
